### DISCUSSION TITLE
convert github-tags from patch to minor

### DISCUSSION THREAD

**Discussion by vquie** [renovatebot/renovate#26485] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.115.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### What is your question?

I'm struggling with updating my kubernetes version in my terraform code. The [terraform eks modul](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest) expects the `cluster_version`  like this: `1.29`. The Github tag I am calling the version from is formatted `v1.29.0`. How do I convert it? I couldn't find any solution.

my tfvars
```terraform
# renovate: datasource=github-tags depName=kubernetes/kubernetes versioning=loose
cluster_version = "1.25"
```

I also tried this
```terraform
# renovate: datasource=github-tags depName=kubernetes/kubernetes versioning=docker
cluster_version = "1.25"
```

my renovate.json
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchPackageNames": [
        "kubernetes/kubernetes"
      ],
      "allowedVersions": "/^(v)?[0-9]+\\.[0-9]+\\.[0-9]+(\\.[0-9]+)?$/"
    }
  ],
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": [
        ".*.tfvars$"
      ],
      "matchStrings": [
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s+cluster_version\\s+=\\s+\"(?<currentValue>.*)\""
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ]
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Use extractVersion

**Reply by vquie** [reply]
Thanks.

```json
    {
      "matchPackageNames": [
        "kubernetes/kubernetes"
      ],
      "extractVersion": "^v(?<version>\\d+\\.\\d+).*$"
    }
```

together with
 
```
# renovate: datasource=github-releases depName=kubernetes/kubernetes versioning=docker
```

did the trick.

**Comment by joshbeard** [comment]
Heads up on this - the latest Kubernetes version isn't necessarily the latest EKS version. Currently, Kubernetes 1.29 is the latest, but EKS supports up to 1.28: https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html

I'm looking to solve the same thing. I'm currently messing with extracting the versions from that link into a JSON file for use with a custom data source. I couldn't find an existing source that's more readable for the EKS versions. Still working on that, myself, but curious if anyone has any ideas on this.

**Reply by joshbeard** [reply]
For the EKS Kubernetes version specifically, I'm generating a local JSON file from the versions scraped from their release page to use as a [custom data source](https://docs.renovatebot.com/modules/datasource/custom/#custom-offline-dependencies).

This works for me:

```json
{
    "customDatasources": {
        "eks": {
            "defaultRegistryUrlTemplate": "file://eks-k8s.json",
            "transformTemplates": []
        }
    },
    "customManagers": [
        {
            "customType": "regex",
            "datasourceTemplate": "custom.eks",
            "fileMatch": ["example-eks-terraform/main.tf"],
            "matchStrings": [
                "cluster_version\\s*=\\s*\"(?<currentValue>\\d+\\.\\d+)\""
            ],
            "depNameTemplate": "eks"
        }
    ],
    "packageRules": [
        {
            "matchDatasources": [
                "custom.eks"
            ],
            "extractVersion": "(?<version>.+)$"
        }
    ]
}
```

`eks-k8s.json`:
```json
{
    "releases": [
        {
            "version": "1.28"
        }
    ],
    "changelogUrl": "https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions-standard.html"
}
```
